### DISCUSSION TITLE
How can I insert a column or row at a specific index in a HeaderedRecord?

### DISCUSSION THREAD

**Discussion by goyaljai1** [clientIO/joint#2776] [Q&A] [answered]
### Introduction

I am having trouble selecting a specific cell with a single click in my application. I would like to be able to select a particular cell easily, and when I do, I want to see buttons that allow me to add a column or a row at that location. This functionality would greatly enhance the user experience by making it more intuitive and efficient to modify the HeaderedRecord.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
This [example](https://docs.jointjs.com/learn/features/shapes/built-in-shapes/records) shows how to select a cell in the record (see Pythagorean triple).

The [DataMapping](https://www.jointjs.com/demos/data-mapping) examples shows how to add new cells to the record via context menu, but it also shows how to render a button next to a specific item (see `Decorator`) .

**Reply by goyaljai1** [reply]
Is there an in-built function to add an entire row or column next to a cell? Also, can we change the column width of a particular column?

**Reply by kumilingus** [reply]
> Is there an in-built function to add an entire row or column next to a cell?

No. You'd have to add an item to each column or add a new column with all rows.

> Also, can we change the column width of a particular column?

No.  This is currently not supported. The size is calculated based on the width of the Record and the number of columns.

---

Here are some alternatives to using Record shapes:
- A custom shape e.g. see this [PR](https://github.com/clientIO/joint/pull/1071/files)
- An integration with [Handsontable](https://www.jointjs.com/demos/spreadsheet-shapes-with-handsontable)
